pkg/server/server.go

```diff
@@ -60,194 +59,171 @@ type Config struct {
 }
 
 type Server struct {
-	Catalog   catalog.Catalog
-	Config    *Config
-	caManager ca.Manager
-	endpoints endpoints.Server
-	svid      svid.Rotator
-
-	m *sync.RWMutex
-	t *tomb.Tomb
+	config Config
+
+	// test hooks
+	hooks struct {
+		newCAManager func(*ca.Config) ca.Manager
```

Comment:
The lifetime of the CA manager was changed so that it only lives for the duration of the Run call.  Run creates it and then passes it around to whomever needs it. I added the hook before I read the tests carefully and thought that i needed a way to inject a different CA manager during tests. Turns out the tests aren't really exercising much, so the hook is actually unnecessary right now. If we add more tests, we'll likely need a way of injecting a fake/mock CA manager impl. For now though, i can remove the hook.